go/runtime/push.go

```diff
@@ -0,0 +1,174 @@
+package runtime
+
+import (
+	"fmt"
+	"io"
+	"math/rand"
+
+	"github.com/estuary/flow/go/labels"
+	"github.com/estuary/protocols/capture"
+	pf "github.com/estuary/protocols/flow"
+	"github.com/sirupsen/logrus"
+	pb "go.gazette.dev/core/broker/protocol"
+	"go.gazette.dev/core/consumer"
+	"go.gazette.dev/core/consumer/protocol"
+	"google.golang.org/grpc"
+)
+
+// TODO(johnny): This implementation builds and I believe it to be correct
+// and what we want, but it's also untested at this time (November 2021).
+// I'm putting it down for the time being, with plans to pick it up as we
+// re-add features around HTTP, WebSocket, and Kafka push ingestion.
+var _ capture.RuntimeServer = (*FlowConsumer)(nil)
+
+func (f *FlowConsumer) Push(stream capture.Runtime_PushServer) error {
```

Comment:
Got this working with an example like [this](https://github.com/estuary/flow/commit/26405051bb28e551b6fbf6802c0986bd7c784dbd). But seems `proxyPush` was not covered in the test example.

Reply:
Awesome! Glad it's working so far.